nano/secure/common.cpp

```diff
@@ -738,6 +754,14 @@ bool nano::vote::validate () const
 	return nano::validate_message (account, hash (), signature);
 }
 
+uint64_t nano::vote::packed_timestamp (uint64_t timestamp, uint8_t duration) const
+{
+	debug_assert ((timestamp & 0xf) == 0 && "Invalid timestamp");
```

Comment:
This check looks wrong. A timestamp could be any number including all zeros at the lowest bits.

Reply:
This check is to force the caller to discard the low 4 bits rather than packed_timestamp discarding them silently. This seemed less confusing but it could be done either way if there's a strong reason to truncate internally.

Reply:
I see, I was expecting it to be the other way and I misread the assert.

I see now that you set the timestamp using this:
`nano::milliseconds_since_epoch () & nano::vote::timestamp_mask `

Personally, I would have gone with the option of silently dropping the timestamp inside the timestamp class just to keep the code simpler. So it would be treating the dropping of the 4 LSBs as a implementation detail and the creator of votes wouldn;t need to do the masking of of LSBs. But there is no right or wrong in these things.

Reply:
Changed this so the caller doesn't need to truncate the bits.